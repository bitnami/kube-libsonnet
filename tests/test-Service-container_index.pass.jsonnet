local bitnami = import "../bitnami.libsonnet";
local kube = import "../kube.libsonnet";

local stack = {
  name:: "test-Service-pass",
  pod: kube.Pod($.name + "-pod") {
    spec+: {
      containers_+: {
        default: kube.Container($.name + "-default") {
          image: "nginx:1.12",
          ports_+: {
            http: { containerPort: 80 },
            metrics: { containerPort: 9099 },
          },
        },
        sidecar: kube.Container($.name + "-sidecar") {
          image: "nginx:1.12",
          ports_+: {
            http_sidecar: { containerPort: 80 },
            metrics: { containerPort: 9099 },
          },
        },
      },
    },
  },
  deploy: kube.Deployment($.name + "-deploy") {
    local this = self,
    spec+: {
      template+: {
        spec+: $.pod.spec,
      },
    },
  },
  service: kube.Service($.name + "-svc") {
    local this = self,
    target_pod: $.deploy.spec.template,
    container_index: 1,
  },
};

stack {
  // Assert we got the 2nd containers port named "http" for service
  assert (stack.service.spec.ports[0].name == "http-sidecar") : "Expected service port http-sidecar.",
}
