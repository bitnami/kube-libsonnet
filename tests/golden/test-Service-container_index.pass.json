{
   "deploy": {
      "apiVersion": "apps/v1",
      "kind": "Deployment",
      "metadata": {
         "annotations": { },
         "labels": {
            "name": "test-Service-pass-deploy"
         },
         "name": "test-Service-pass-deploy"
      },
      "spec": {
         "minReadySeconds": 30,
         "replicas": 1,
         "revisionHistoryLimit": 10,
         "selector": {
            "matchLabels": {
               "name": "test-Service-pass-deploy"
            }
         },
         "strategy": {
            "rollingUpdate": {
               "maxSurge": "25%",
               "maxUnavailable": "25%"
            },
            "type": "RollingUpdate"
         },
         "template": {
            "metadata": {
               "annotations": { },
               "labels": {
                  "name": "test-Service-pass-deploy"
               }
            },
            "spec": {
               "containers": [
                  {
                     "args": [ ],
                     "env": [ ],
                     "image": "nginx:1.12",
                     "imagePullPolicy": "IfNotPresent",
                     "name": "test-Service-pass-default",
                     "ports": [
                        {
                           "containerPort": 80,
                           "name": "http"
                        },
                        {
                           "containerPort": 9099,
                           "name": "metrics"
                        }
                     ],
                     "stdin": false,
                     "tty": false,
                     "volumeMounts": [ ]
                  },
                  {
                     "args": [ ],
                     "env": [ ],
                     "image": "nginx:1.12",
                     "imagePullPolicy": "IfNotPresent",
                     "name": "test-Service-pass-sidecar",
                     "ports": [
                        {
                           "containerPort": 80,
                           "name": "http-sidecar"
                        },
                        {
                           "containerPort": 9099,
                           "name": "metrics"
                        }
                     ],
                     "stdin": false,
                     "tty": false,
                     "volumeMounts": [ ]
                  }
               ],
               "imagePullSecrets": [ ],
               "initContainers": [ ],
               "terminationGracePeriodSeconds": 30,
               "volumes": [ ]
            }
         }
      }
   },
   "pod": {
      "apiVersion": "v1",
      "kind": "Pod",
      "metadata": {
         "annotations": { },
         "labels": {
            "name": "test-Service-pass-pod"
         },
         "name": "test-Service-pass-pod"
      },
      "spec": {
         "containers": [
            {
               "args": [ ],
               "env": [ ],
               "image": "nginx:1.12",
               "imagePullPolicy": "IfNotPresent",
               "name": "test-Service-pass-default",
               "ports": [
                  {
                     "containerPort": 80,
                     "name": "http"
                  },
                  {
                     "containerPort": 9099,
                     "name": "metrics"
                  }
               ],
               "stdin": false,
               "tty": false,
               "volumeMounts": [ ]
            },
            {
               "args": [ ],
               "env": [ ],
               "image": "nginx:1.12",
               "imagePullPolicy": "IfNotPresent",
               "name": "test-Service-pass-sidecar",
               "ports": [
                  {
                     "containerPort": 80,
                     "name": "http-sidecar"
                  },
                  {
                     "containerPort": 9099,
                     "name": "metrics"
                  }
               ],
               "stdin": false,
               "tty": false,
               "volumeMounts": [ ]
            }
         ],
         "imagePullSecrets": [ ],
         "initContainers": [ ],
         "terminationGracePeriodSeconds": 30,
         "volumes": [ ]
      }
   },
   "service": {
      "apiVersion": "v1",
      "kind": "Service",
      "metadata": {
         "annotations": { },
         "labels": {
            "name": "test-Service-pass-svc"
         },
         "name": "test-Service-pass-svc"
      },
      "spec": {
         "ports": [
            {
               "name": "http-sidecar",
               "port": 80,
               "targetPort": 80
            }
         ],
         "selector": {
            "name": "test-Service-pass-deploy"
         },
         "type": "ClusterIP"
      }
   }
}
